web/src/portico/signup.ts

```diff
@@ -21,11 +21,15 @@ $(() => {
         // Reset the state of the password strength bar if the page
         // was just reloaded due to a validation failure on the backend.
         password_quality($password_field.val()!, $("#pw_strength .bar"), $password_field);
-
+        let timer: ReturnType<typeof setTimeout>;
         $password_field.on("input", function () {
             // Update the password strength bar even if we aren't validating
             // the field yet.
-            password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
+            clearTimeout(timer);
+            timer = setTimeout(
+                () => password_quality($(this).val()!, $("#pw_strength .bar"), $(this)),
+                500,
+            );
```

Comment:
Why not use `_.debounce` or similar?

Reply:
@timabbott I've made the changes.
Also I had to remove the jquery validation from keyup events in the form elements since it executed separately from the debounced calls, and rendered the debouncing useless. I was hoping you'd review and tell if it's fine or not.